SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslationManager.cpp

```diff
@@ -76,42 +92,111 @@ void SpatialVirtualWorkerTranslationManager::AuthorityChanged(const Worker_Compo
 	}
 }
 
+void SpatialVirtualWorkerTranslationManager::SetKnownServerSystemEntities(TArray<Worker_EntityId> ServerSystemEntities)
+{
+	KnownServerSystemEntities = ServerSystemEntities;
+}
+
+void SpatialVirtualWorkerTranslationManager::OnWorkerDisconnected(const Worker_EntityId DisconnectedSystemEntityId)
+{
+	// If we finish this loop without hitting anything, we assume it's a client that disconnected.
+	for (auto& VirtualWorkerInfo : VirtualToPhysicalWorkerMapping)
+	{
+		if (VirtualWorkerInfo.Value.ServerSystemWorkerEntity == DisconnectedSystemEntityId)
+		{
+			UE_LOG(LogSpatialVirtualWorkerTranslationManager, Warning, TEXT("VTM identified disconnected server %s"),
+				   *VirtualWorkerInfo.Value.PhysicalWorkerName);
+			UE_LOG(LogSpatialVirtualWorkerTranslationManager, Warning, TEXT(" - virtual worker %ld"), VirtualWorkerInfo.Key);
+			UE_LOG(LogSpatialVirtualWorkerTranslationManager, Warning, TEXT(" - partition %lld"), VirtualWorkerInfo.Value.PartitionId);
+			UE_LOG(LogSpatialVirtualWorkerTranslationManager, Warning, TEXT(" - system entity %lld"),
+				   VirtualWorkerInfo.Value.ServerSystemWorkerEntity);
+
+			ReclaimCrashedVirtualWorker(VirtualWorkerInfo.Key);
+		}
+	}
+}
+
+void SpatialVirtualWorkerTranslationManager::CleanupTranslatorMappingAfterAuthorityChange()
+{
+	TArray<VirtualWorkerId> CrashedVirtualWorkers;
```

Comment:
I feel like using "crashed" here (and elsewhere) is too presumptive. It's just "missing"